### PR TITLE
Add github workflow running jenkins job

### PR DESCRIPTION
**Pull request by sylwiaszunejko**
This workflow pushes scylla-bench docker image using gocql version from the PR and triggers jenkins job that uses this image of scylla-bench.

### CODE REVIEW

**Comment by sylwiaszunejko** [comment]
New workflow won't be running in CI in this PR, because it uses pull_request_target event. In this PR we can see it running: https://github.com/scylladb/gocql/pull/263

Things to be changed:
* change jenkins_client fork to scylladb
* change user from me to some official jenkins user
* move jenkins job to official location

**Comment by dkropachev** [comment]
@sylwiaszunejko , it is fialing due to the timeout https://github.com/scylladb/gocql/actions/runs/10928426706/job/30337696759?pr=263

**Comment by sylwiaszunejko** [comment]
> @sylwiaszunejko , it is fialing due to the timeout https://github.com/scylladb/gocql/actions/runs/10928426706/job/30337696759?pr=263

I know I just trigger two jobs too close to each other then trying different images, that is why

File: .github/workflows/extended-ci-longevity-large-partition-asymmetric-cluster-3h-test.yml
Comment (Lorak-mmk) [inline .github/workflows/extended-ci-longevity-large-partition-asymmetric-cluster-3h-test.yml:48]: As far as I can tell, goland doesn't allow code execution during build, so it should not be possible to hijack CI using some malicious code.
The problem I see is that the result of such build is pushed to dockerhub and is available under scylladb org - are we ok with having potentially malicious images there?
Jenkins: the same problem. Is our infrastructure ready to execute untrusted code? I highly doubt it.

Comment (dkropachev) [reply .github/workflows/extended-ci-longevity-large-partition-asymmetric-cluster-3h-test.yml:48]: it is pushed to a repo that is designated to this specific purpose, so chance that someone accidentially run this image is effectively zero.
This image is going to be run on a temporary instance which has access to the db nodes only, beside generic internet access, so you can consider it pretty safe.

Comment (dkropachev) [reply .github/workflows/extended-ci-longevity-large-partition-asymmetric-cluster-3h-test.yml:48]: > Are you sure we are not putting some SSH keys / env variables etc on those nodes that should not be publicly available? Could someone from QA confirm it? @fruch @roydahan @soyacz

100% loader is getting controlled by sct-runner, which has all the keys, loaders them selves have nothing.

Comment (fruch) [reply .github/workflows/extended-ci-longevity-large-partition-asymmetric-cluster-3h-test.yml:48]: 1) those machines are not publicly available 
2) users we don't trust won't be able to execute this action, we'll need to actively add them as contributors to the project for this CI part to be working for them (as we do in lots of other public projects, i.e. scylla-core)

Comment (Lorak-mmk) [reply .github/workflows/extended-ci-longevity-large-partition-asymmetric-cluster-3h-test.yml:48]: So a maintainer will have to approve each CI run from outside user?

Comment (fruch) [reply .github/workflows/extended-ci-longevity-large-partition-asymmetric-cluster-3h-test.yml:48]: > So a maintainer will have to approve each CI run from outside user?

Each external user, adding them as contributers.

maybe there's a way to configure the action to need approval on each run, but it's on the actions configurations

Comment (Lorak-mmk) [reply .github/workflows/extended-ci-longevity-large-partition-asymmetric-cluster-3h-test.yml:48]: > > So a maintainer will have to approve each CI run from outside user?
> 
> Each external user, adding them as contributers.

After being added as contributors they wouldn't need approval anymore? In that case account overtake of such user would be dangerous for us.

> 
> maybe there's a way to configure the action to need approval on each run, but it's on the actions configurations

It is possible to require CI approval for all runs from external contributors:
image

I don't know any way to require this for a single job.

Comment (dkropachev) [reply .github/workflows/extended-ci-longevity-large-partition-asymmetric-cluster-3h-test.yml:48]: There is a way via deployment environment, but us realy inconvinient, since code is ran in isolated environment I don't see any reason on doing that.

**Comment by sylwiaszunejko** [comment]
After some fixes on scylla-bench side we have a successful run https://jenkins.scylladb.com/job/scylla-staging/job/sylwia.szunejko/job/longevity-large-partition-asymmetric-cluster-3h-test/55/ 
@fruch should we now define an official location for those jobs?

**Comment by sylwiaszunejko** [comment]
@roydahan What should be the structure of this official location? Drivers ->  Gocql -> longevity-large-partition-asymmetric-cluster-3h-test?

**Comment by dkropachev** [comment]
> @roydahan What should be the structure of this official location? Drivers -> Gocql -> longevity-large-partition-asymmetric-cluster-3h-test?

And do we need to run asymmetric-cluster test ? or it is better to have regular 3h longevity for sake of stability ?

**Comment by sylwiaszunejko** [comment]
@roydahan Should the next steps be decided on SCT side e.g. which test to run and what should be the official location for those jenkins jobs?

**Comment by dkropachev** [comment]
@sylwiaszunejko , please take a look at https://jenkins.scylladb.com/job/scylla-drivers/job/gocql/job/advanced-ci-cd/job/longevity-large-partition-asymmetric-cluster-3h-test/

**Comment by sylwiaszunejko** [comment]
> @sylwiaszunejko , please take a look at `job/scylla-drivers/job/gocql/job/advanced-ci-cd/job/longevity-large-partition-asymmetric-cluster-3h-test/`

shouldn't it be named `extended-ci` instead of `advanced-ci-cd` for consistency with the label and dockerhub repo? https://hub.docker.com/r/scylladb/gocql-extended-ci/tags

**Comment by dkropachev** [comment]
> > @sylwiaszunejko , please take a look at `job/scylla-drivers/job/gocql/job/advanced-ci-cd/job/longevity-large-partition-asymmetric-cluster-3h-test/`
> 
> shouldn't it be named `extended-ci` instead of `advanced-ci-cd` for consistency with the label and dockerhub repo? https://hub.docker.com/r/scylladb/gocql-extended-ci/tags

Done

**Comment by sylwiaszunejko** [comment]
So it looks like we have an official location for a job, next thing would be to have a designated user for it, is there some generic one we can use or who could create such?

File: .github/workflows/extended-ci-longevity-large-partition-asymmetric-cluster-3h-test.yml
Comment (roydahan) [inline .github/workflows/extended-ci-longevity-large-partition-asymmetric-cluster-3h-test.yml:44]: @yaronkaikov do we have a jenkins user we can for the CI?

File: .github/workflows/extended-ci-longevity-large-partition-asymmetric-cluster-3h-test.yml
Comment (dkropachev) [reply .github/workflows/extended-ci-longevity-large-partition-asymmetric-cluster-3h-test.yml:42]: > I'm unsure if we should fix the scylla_version to something specific. On one hand, it can make the CI more predictable (not sure if stable, but quite repeatable). On the other hand, at some point we need to go back and update it and possibly face new issues.

having `master` there is not an option.
We can do same thing we do for `cpp-rust-driver` - get last release and run on it.

> BTW, there is an option to set here "scylla_version: 6.1" so it will take all scylla 6.1 patches. Another option is to use an enterprise release like 2024.1 who is LTS and won't require us to update it very often (will automatically pick the latest patch release).

From one side using OSS will bring top feature to the test as fast as possible.
On other side using enterprise LTS will bring more stability.
It looks like we need both ))

> Regarding email_recipent, can we set it to the author of the PR (as long as it internal author?) - not a must.

It is doable, but it will expose ip addresses, urls of our test infrastructure, if we are ok with it then sure, why not.
But we also should have scylla-drivers@scylladb.com as CC.

> Lastly, I noticed that the Jenkins job is set to master, which again brings back the question of stability vs being updated automatically.

Agree, it is better to target either `branch-6.1` or `branch-2023` or particular commit from master

Comment (fruch) [reply .github/workflows/extended-ci-longevity-large-partition-asymmetric-cluster-3h-test.yml:42]: > > I'm unsure if we should fix the scylla_version to something specific. On one hand, it can make the CI more predictable (not sure if stable, but quite repeatable). On the other hand, at some point we need to go back and update it and possibly face new issues.
> 
> having `master` there is not an option.
> We can do same thing we do for `cpp-rust-driver` - get last release and run on it.
> 
> > BTW, there is an option to set here "scylla_version: 6.1" so it will take all scylla 6.1 patches. Another option is to use an enterprise release like 2024.1 who is LTS and won't require us to update it very often (will automatically pick the latest patch release).
> 
> From one side using OSS will bring top feature to the test as fast as possible.
> On other side using enterprise LTS will bring more stability.
> It looks like we need both ))
> 
> > Regarding email_recipent, can we set it to the author of the PR (as long as it internal author?) - not a must.
> 
> It is doable, but it will expose ip addresses, urls of our test infrastructure, if we are ok with it then sure, why not.
> But we also should have scylla-drivers@scylladb.com as CC.

Nothing secret in the emails
Also those should run only by trusted users (with some definition of trust), isn't it ?

> 
> > Lastly, I noticed that the Jenkins job is set to master, which again brings back the question of stability vs being updated automatically.
> 
> Agree, it is better to target either `branch-6.1` or `branch-2023` or particular commit from master
> 

release branch do get backports, on the other hand they might not have all of the fixes features for those test (not to mention we'll need to backport specific things unrelated to a release need)

A specific master commit, is also problematic, when one will update it ? what happens when it breaks and stops working ?

File: .github/workflows/extended-ci-longevity-large-partition-asymmetric-cluster-3h-test.yml
Comment (yaronkaikov) [reply .github/workflows/extended-ci-longevity-large-partition-asymmetric-cluster-3h-test.yml:44]: I think you can use `scylladb-promoter` but we should probably also set the `token` as `GITHUB_SECRETS`

Comment (fruch) [reply .github/workflows/extended-ci-longevity-large-partition-asymmetric-cluster-3h-test.yml:44]: Let's have those creds at org level, we'll need them in more projects that would follow

Comment (yaronkaikov) [reply .github/workflows/extended-ci-longevity-large-partition-asymmetric-cluster-3h-test.yml:44]: Done, should use `JENKINS_USERNAME` and `JENKINS_TOKEN`, it's organization level, for now only for the `scylladb/gocql` repo. Let's make sure it's working then we can add additional repos

Comment (sylwiaszunejko) [reply .github/workflows/extended-ci-longevity-large-partition-asymmetric-cluster-3h-test.yml]: For some reason triggering job does not work with the credentials provided by @yaronkaikov  https://github.com/scylladb/gocql/actions/runs/11120234541/job/30896983496?pr=263 my guess is that JENKINS_TOKEN secret on the org level has the same name as token we used for my username, @dkropachev could you maybe remove this repo level token?

Comment (dkropachev) [reply .github/workflows/extended-ci-longevity-large-partition-asymmetric-cluster-3h-test.yml]: Done

**Comment by sylwiaszunejko** [comment]
https://github.com/scylladb/gocql/actions/runs/11157393428/job/31011588519?pr=263 I am not sure why the second workflow gets interrupted even though the timeout has not passed yet and the job is still running

**Comment by sylwiaszunejko** [comment]
@dkropachev is https://github.com/scylladb-actions/jenkins-client an official location for this action?

**Comment by sylwiaszunejko** [comment]
> what branch should Jenkins job target
is running asymmetric-cluster test the right choice or should we choose different one?

@roydahan @fruch could you help with this questions?

**Comment by dkropachev** [comment]
> @dkropachev is https://github.com/scylladb-actions/jenkins-client an official location for this action?

It is not, but it shouldn't matter since it is an GitHub action.

**Comment by roydahan** [comment]
> > what branch should Jenkins job target
> 
> is running asymmetric-cluster test the right choice or should we choose different one?
> 
> 
> 
> @roydahan @fruch could you help with this questions? 

I'm working on stabilizing another shorter test for CI (1h test that is based on this test).
+
My comment from a different related issue:

I'm not sure what you mean by "need to manually create new job".

I also don't think this mechanism is necessary.

IMO the CI should use 2024.1 as Scylla_version and the SCT branch is branch-2024.1.
This promises to test on latest 2024.1 which LTS and make sure there are no regressions.

In addition, we can have another CI job if we think is necessary that points to "scylla_version: master:latest".
This job can be used with SCT branch master and should be used in order to test our driver with the latest and greatest Scylla features.
It won't be as stable as the first one, but it will be a good addition to get an additional data point on client stability.

**Comment by sylwiaszunejko** [comment]
> > > what branch should Jenkins job target
> > 
> > 
> > is running asymmetric-cluster test the right choice or should we choose different one?
> > @roydahan @fruch could you help with this questions?
> 
> I'm working on stabilizing another shorter test for CI (1h test that is based on this test). + My comment from a different related issue:
> 
> I'm not sure what you mean by "need to manually create new job".
> 
> I also don't think this mechanism is necessary.
> 
> IMO the CI should use 2024.1 as Scylla_version and the SCT branch is branch-2024.1. This promises to test on latest 2024.1 which LTS and make sure there are no regressions.
> 
> In addition, we can have another CI job if we think is necessary that points to "scylla_version: master:latest". This job can be used with SCT branch master and should be used in order to test our driver with the latest and greatest Scylla features. It won't be as stable as the first one, but it will be a good addition to get an additional data point on client stability.

I thought about the scenario when 2024 is no longer the latest lts and we want to upgrade, then we want to change the configuration manually, and the same if we want to have second job for latest oss minor version

**Comment by roydahan** [comment]
Tested the new job for CI with various nemesis_seeds (in order to fix it with a specific one later): https://argus.scylladb.com/tests/scylla-cluster-tests/1adffeeb-28ac-432c-a6f8-8477b70fa557
For now it seems like the CI job is very stable (suspiciously too stable).

**Comment by dkropachev** [comment]
> Tested the new job for CI with various nemesis_seeds (in order to fix it with a specific one later): https://argus.scylladb.com/tests/scylla-cluster-tests/1adffeeb-28ac-432c-a6f8-8477b70fa557 For now it seems like the CI job is very stable (suspiciously too stable).

Great, what is the next step for the job ? do you want to merge it ? should we finally merge this PR pointing to a job with temporary sct repo/branch ?

**Comment by sylwiaszunejko** [comment]
I have changed the logic to use https://github.com/scylladb-actions/get-version/tree/main to get scylladb latest version

**Comment by roydahan** [comment]
I was working to add more partitions to the test so the workload will be more balanced across the shards, but now I need to stablize it a bit since the load is higher now.
I'll try to get done with it by early next week, so we can use it from the beginning.

**Comment by sylwiaszunejko** [comment]
@roydahan should we use new test as https://github.com/scylladb/scylla-cluster-tests/pull/9137 is merged?

**Comment by roydahan** [comment]
Yes, this is the new job: https://jenkins.scylladb.com/view/scylla-drivers/job/scylla-drivers/job/gocql/job/extended-ci/job/longevity-large-partitions-with-network-nemesis-1h-test/

**Comment by sylwiaszunejko** [comment]
It keeps failing on provisioning resources, when I changed the instance type from spot to on_demand it works.
@roydahan is it a good idea to change instant type to on_demand as a default?

**Comment by roydahan** [comment]
It's probably a transient issue with spot availability.
IIRC, we should have an automatic fallback in this case, I can check that.

Another option is to trigger it for now with on_demand.

**Comment by roydahan** [comment]
Sent https://github.com/scylladb/scylla-cluster-tests/pull/9180 to add the fallback by default.

**Comment by sylwiaszunejko** [comment]
@dkropachev with on_demand the test works https://github.com/scylladb/gocql/pull/263/checks, should we merge this PR?

**Review by dkropachev — APPROVED** [review]

**Pull request merged**
